src/cargo/core/compiler/custom_build.rs

```diff
@@ -690,7 +690,17 @@ impl BuildOutput {
                 continue;
             }
             let data = match iter.next() {
-                Some(val) => val,
+                Some(val) => {
+                    if val.starts_with(":") {
+                        // Line started with `cargo::`.
+                        bail!("invalid output in {}: `{}`\n\
+                            Expected a line with `cargo:key=value`, \
+                            but found `cargo::` instead.\n\
+                            See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script \
+                            for more information about build script outputs.", whence, line);
+                    }
```

Comment:
Should we make a comment about `cargo::` is reserved for future use or some way to indicate that there is a version mismatch between the build script and cargo, whether its because no MSRV was specified or someone was reading too new of documentation?

Reply:
Good suggestion. Thank you.

Reply:
Sorry, I meant to the end-user in the error message.

Its not that this is invalid syntax but that we've reserved the syntax for future use.  Explaining that to a user will help them realize how to resolve the error (upgrade or not use it) vs being left without a way to resolve it.

Reply:
Oh, got it. Updated.